ui/src/main/webapp/admin-tools/wcpm/whitelist.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+
+import { fromJS, Map } from 'immutable'
+
+import { List, ListItem } from 'material-ui/List'
+
+import { Header, EditRegion, ConfirmationPanel, ServerErrors } from './components'
+
+import Description from 'components/Description'
+
+import EditableList from './editable-list'
+
+const Layout = ({ children }) => (
+  <div>
+    <div style={{ padding: 8, textAlign: 'center' }}>
+      <Header>Whitelisted Contexts</Header>
+      <Description>
+        List of context paths that will not use security. Note that sub-contexts to
+        ones listed here will also skip security, unless authentication types
+        are provided for it. For example: if /foo is listed here, then /foo/bar
+        will also not require any sort of authentication. However, if /foo is
+        listed and /foo/bar has authentication types provided in the
+        'Authentication Types' field, then that more specific policy will be
+        used.
+      </Description>
+    </div>
+    {children}
+  </div>
+)
+
+const WhiteListView = ({ whitelisted, onEdit }) => (
+  <EditRegion onEdit={() => onEdit(fromJS(whitelisted))}>
+    <Layout>
+      <List>
+        {whitelisted.map((path, i) => <ListItem disabled key={i} primaryText={path} />)}
+      </List>
+    </Layout>
+  </EditRegion>
+)
+
+export const WhitelistEdit = ({ whitelisted = [], errors, serverErrors = {}, onUpdate, onCancel, onSave }) => (
+  <Layout>
+    <div style={{ padding: 8 }}>
+      <EditableList
+        hintText='Add New Path'
+        list={whitelisted}
+        errors={errors}
+        onChange={({ value, index }) => onUpdate([index], value === '' ? undefined : value)} />
+      <ConfirmationPanel onCancel={onCancel} onSave={() => onSave(whitelisted)} />
+      <ServerErrors errors={serverErrors.all} />
+    </div>
+  </Layout>
+)
+
+const isValidContextPath = (path) => {
+  return /^(\/[A-Za-z0-9-._~:/?#[\]@!$&'()*+,;=`.%]*)+$/g.test(path)
```

Comment:
I'm not sure how many places we're going to end up using this, but it might be worthwhile to stick this in `lib` somewhere